collector/textfile.go

```diff
@@ -63,17 +64,27 @@ type textFileCollector struct {
 func newTextFileCollectorFlags(app *kingpin.Application) {
 	textFileDirectory = app.Flag(
 		FlagTextFileDirectory,
-		"Directory to read text files with metrics from.",
-	).Default(getDefaultPath()).String()
+		"DEPRECATED: Use --collector.textfile.directories",
+	).Default("").String()
```

Comment:
I feel we should add `.Hidden()` here to discourage use of the flag for new users, similar to the deprecated flags in the `process` and other exporters.